### PR TITLE
Modify perf test

### PR DESCRIPTION
**Pull request by lionkunonly**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by lionkunonly** [comment]
#332

File: modules/js/perf/perf_imgproc/perf_cvtcolor.js [hunk 1]
```diff
@@ -4,20 +4,36 @@ if (isNodeJs) {
...
 
-cv.onRuntimeInitialized = () => {
+var cvSize
+
+async function main() {
+  if (cv instanceof Promise) {
+    cv = await cv
+  }
+
+  cvSize = {
```
Comment (huningxin) [inline modules/js/perf/perf_imgproc/perf_cvtcolor.js]: Put cvSize definition in every files are duplicated. I understand the calling to cv.Size is not valid before cv promise is resolved. Could you please create a function inside base.js (e.g. createCvSize) that returns the cvSize? You then can call that function after await.

File: modules/js/perf/perf_imgproc/perf_cvtcolor.js [hunk 2]
```diff
@@ -4,20 +4,36 @@ if (isNodeJs) {
...
 } else {
   var paramsElement = document.getElementById('params');
   var runButton = document.getElementById('runButton');
   var logElement = document.getElementById('log');
 }
 
-cv.onRuntimeInitialized = () => {
+var cvSize
+
+async function main() {
```
Comment (huningxin) [inline modules/js/perf/perf_imgproc/perf_cvtcolor.js]: The cv.onRuntimeInitialized path is still needed if cv is not a promise. So I suppose the code logic should be:
```js
if (cv instanceof Promise) {
  cv = await cv;
  main();
} else {
  cv.onRuntimeInitialized = main;
}
```

**Review by huningxin — APPROVED** [review]
PR looks good. Thanks!

**Pull request merged**